astropy/modeling/core.py

```diff
@@ -2349,6 +2351,63 @@ def __init__(self, op, left, right, name=None, inverse=None):
                                               self.left.inverse,
                                               inverse=self)
+        elif op == 'fix_inputs':
+            if not isinstance(left, Model):
+                raise ValueError('First argument must be an instance of an astropy Model')
+            if not isinstance(right, dict):
+                raise ValueError('expecting dictionary for second argument')
+
+            # Dict keys must match either possible indices
+            # for model on left side,
+            # or names for inputs.
+            self.n_inputs = left.n_inputs - len(right)
+            self.outputs = left.outputs
+            self.n_outputs = left.n_outputs
+            newinputs = list(left.inputs)
+            keys = right.keys()
+            input_ind = []
+            for key in keys:
+                if isinstance(key, int):
+                    if key >= left.n_inputs or key < 0:
+                        raise ValueError(
+                            'substitution key integer value '
+                            'not among possible input choices')
+                    else:
+                        if key in input_ind:
+                            raise ValueError("Duplicate specification of "
+                                             "same input (index/name)")
+                        else:
+                            input_ind.append(key)
+                elif isinstance(key, str):
+                    if key not in left.inputs:
+                        raise ValueError(
+                            'Substitution key string not among possible '
+                            'input choices')
+                    # Check to see it doesn't match positional
+                    # specification.
+                    ind = left.inputs.index(key)
+                    if ind in input_ind:
+                        raise ValueError("Duplicate specification of "
+                                         "same input (index/name)")
+                    else:
+                        input_ind.append(ind)
+            # Remove substituted inputs
+            input_ind.sort()
+            input_ind.reverse()
+            for ind in input_ind:
+                del newinputs[ind]
+            self.inputs = tuple(newinputs)
+            # Now check to see if the input model has bounding_box defined.
+            # If so, remove the appropriate dimensions and set it for this
+            # instance.
+            try:
+                bounding_box = self.left.bounding_box
+                self._fix_input_bounding_box(input_ind)
+            except NotImplementedError:
+                print('didnt find bounding box')
+           # Call the hook if it exists
+            if self._fix_input_hook is not None:
+                self._fix_input_hook(input_ind)
         else:
             #raise ModelDefinitionError('Illegal operator: ', self.op)
             raise ModelDefinitionError(
```

Comment:
Isn't the commented out `Error` what this `else` statement is about? 
Isn't the current error  for `+`, `-`, `*` ... operators.

Reply:
Do you mean just remove the commented out raise?

Reply:
No, I think the commented out error is the correct one for this `else` statement. The sequence is
```
if op in ['+', '-', '*', '/', '**'] ...:
   ...
elif op == '&':
    ....
elif op == '|':
    ....
elif op == 'fix_inputs':
    ....
else:
    **#Raise an error about Illegal operator**
```

As it stands the error message is misleading. For example
```
Gaussian1D() @ Gaussian1D()
TypeError: unsupported operand type(s) for @: 'Gaussian1D' and 'Gaussian1D'

```